scripts/veloxbench/veloxbench/cpp_micro_benchmarks.py

```diff
@@ -0,0 +1,121 @@
+COMMON_OPTIONS = {
+    "suite-filter": {
+        "default": None,
+        "type": str,
+        "help": "Regex filtering benchmark suites.",
+    },
+}
+
+
+def get_run_command(filename, options):
+    # TODO: we'll probably want to unify these runner scripts
+    # Must run `make benchmarks-basic-build` before this
+    # TODO: deal with the paths better (especially to benchmarks/basic)
+    command = [
+        "../../../scripts/benchmark-runner.py",
+        "run",
+        "--path",
+        "../../../_build/release/velox/benchmarks/basic/",
+        "--dump-path",
+        filename,
+    ]
+
+    # TODO: extend cpp micro benchmarks to allow for iterations
+    iterations = options.get("iterations", None)
+    if iterations:
+        raise NotImplementedError()
+
+    return command
+
+
+def _parse_benchmark_name(full_name):
+    # TODO: Do we need something more complicated?
+    # https://github.com/ursacomputing/benchmarks/blob/033eee0951adbf41931a2de95caccbac887da6ff/benchmarks/cpp_micro_benchmarks.py#L86-L103
+    return {"name": full_name}
+
+
+@conbench.runner.register_benchmark
+class RecordCppMicroBenchmarks(_benchmark.Benchmark):
+    """Run the Velox C++ micro benchmarks."""
+
+    external = True
+    name = "cpp-micro"
+    # see TODO about COMMON_OPTIONS above, set an empty dict for now
+    options = copy.deepcopy(COMMON_OPTIONS)
+    options.update(**RUN_OPTIONS)
+    description = "Run the Velox C++ micro benchmarks."
+    iterations = 1
+    flags = {"language": "C++"}
+
+    def run(self, **kwargs):
+        with tempfile.TemporaryDirectory() as result_dir:
+            run_command = get_run_command(result_dir, kwargs)
+            self.execute_command(run_command)
+            # iterate through files to make the suites
+            with os.scandir(result_dir) as result_files:
+                for result_file in result_files:
+                    suite = result_file.name.removesuffix(".json")
+                    with open(result_file.path, "r") as f:
+                        results = json.load(f)
+                    self.conbench.mark_new_batch()
+                    for result in results:
+                        yield self._record_result(suite, result, kwargs)
+
+    def _record_result(self, suite, result, options):
+        info, context = {}, {"benchmark_language": "C++"}
+        # TODO: we should really name these in the json and not rely on position!
+        tags = _parse_benchmark_name(result[1])
+        name = tags.pop("name")
+        tags["suite"] = suite
+        tags["source"] = self.name
+
+        values = self._get_values(result)
+
+        return self.record(
+            values,
+            tags,
+            info,
+            context,
+            options=options,
+            output=result,
+            name=name,
+        )
+
+    def _get_values(self, result):
+        return {
+            "data": [result[2]],
+            # TODO: should we get units from the results?
```

Comment:
what do you mean? I have the impression that all values are reported by default in ns, no?

Reply:
Ah, ok I'll change these to ns then. 

It looks like the console output for folly includes both time/iter and iters/s, but the json that's written out only includes time/iter. I looked briefly in the folly docs but couldn't find a way to include both, would we ever care about iter/s? Probably not since it's basically 1/(time/iter), but just checking.

For the Arrow benchmarks we have a number of benchmarks that measure throughput, so that gets recorded (alongside the time) and treated slightly differently. Is that something we might need in Velox too?